connectors/src/lib/sync_status.ts

```diff
@@ -1,18 +1,31 @@
 import { Connector, ModelId } from "@connectors/lib/models";
 import { Err, Ok, Result } from "@connectors/lib/result";
-import { ConnectorSyncStatus } from "@connectors/types/connector";
+import {
+  ConnectorErrorType,
+  ConnectorSyncStatus,
+} from "@connectors/types/connector";
 
-async function syncFinished(
-  connectorId: ModelId,
-  status: ConnectorSyncStatus,
-  finishedAt: Date
-): Promise<Result<void, Error>> {
+async function syncFinished({
+  connectorId,
+  status,
+  finishedAt,
+  errorMessage,
```

Comment:
Let’s make it a specific error type here, no need to have two strings in the signature of this function

Reply:
And do it | null